tests/testthat/test-community.R

```diff
@@ -0,0 +1,443 @@
+test_that("community detection functions work", {
+  withr::local_seed(42)
+
+  cluster_algos <- list(
+    "cluster_edge_betweenness", "cluster_fast_greedy",
+    "cluster_label_prop", "cluster_leading_eigen",
+    "cluster_louvain", "cluster_spinglass", "cluster_walktrap"
+  )
+  if (has_glpk()) cluster_algos <- c(cluster_algos, list("cluster_optimal"))
+
+  karate <- make_graph("Zachary")
+
+  for (algo in cluster_algos) {
+    cluster_algo <- get(algo)
```

Comment:
I find this awkward but it might be due to my never using `get`.

I would be more at ease with `do.call(algo, karate)` if that makes sense.

Reply:
I agree that `do.call` is better here

Reply:
fyi it must be `do.call(algo,list(karate))` because `karate` is a list by itself and then interpreted as a list of args by `do.call`